cypress/e2e/gui/mudarTema.cy.js

```diff
@@ -0,0 +1,31 @@
+/// <reference types="cypress" />
+
+describe('Validar Cabeçalho e Mudar o Tema', () => {
+    beforeEach(() => {
+        cy.visit('http://localhost:3000/')
+
+    });
+
+    it('Tema escuro para claro', () => {
+
+        cy.get('#theme-toggle-button')
+            .should('be.visible')
+            .click()
+
+        cy.get('h1').should('exist') // Verifica que o h1 existe dentro do cabeçalho
```

Comment:
Em vez de verificar pela existência do elemento no DOM, verifique por sua visibilidade.

Um elemento pode estar no DOM e não estar visível, portanto, verifique sempre que possível pela visibilidade.

Além disso, verifique não só pela tag HTML, mas também por seu texto.
Dessa forma, garante-se que o texto correto é exibido para os/as usuários/as do software.

Ex.: `cy.contains('h1', 'EngageSphere').should('be.visible')`

Reply:
obrigada pela dica